### PR TITLE
Date detection module with tests and databases required for that

### PR DESCRIPTION
**Pull request by bhagyakjain**
Added -
1. date_detection.py detects the columns that contains dates and also categorizes it into the 3 categories.
2. test_date_detection.py tests the date_detection.py module.

This code works well for common cases available on kaggle.

### CODE REVIEW

**Comment by bhagyakjain** [comment]
I split the the single files into 3 based on the function they do. Also added some comments that make more sense.

File: date_detection/date_column_type_detection.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import pandas
+
+def get_column_type_and_order(column):
+    """
+    This function classifies the column into the 3 types based on
+    the values in it. It also returns the day_first = Bool for the column.
+
+    Args - column - type list
+    Returns - dict - {type_str, day_first}}
+                type_str = 'Consistent' / 'all ambiguous' / 'Inconsistent'
```
Comment (arijitde-goog) [inline date_detection/date_column_type_detection.py:29]: Can we use an enum here instead of strings?

Comment (arijitde-goog) [reply date_detection/date_column_type_detection.py:29]: Alternatively, the strings should be like enums eg: 'CONSINTENT', 'ALL_AMBIGUOUS', 'INCONSISTENT'

Comment (bhagyakjain) [reply date_detection/date_column_type_detection.py]: Done

**Pull request merged**